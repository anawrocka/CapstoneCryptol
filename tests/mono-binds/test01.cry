module test01 where

a x = f x
  where
  f y = x # y
