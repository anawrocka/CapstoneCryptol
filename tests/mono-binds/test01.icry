:set debug=on
:set mono-binds=off
:l test01.cry
:set mono-binds=on
:l test01.cry
