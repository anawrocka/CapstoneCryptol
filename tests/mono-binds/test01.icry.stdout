Loading module Cryptol
Loading module Cryptol
Loading module test01
module test01
import Cryptol
/* Not recursive */
test01::a : {a, b} (fin a) => [a]b -> [2 * a]b
test01::a = \{a, b} (fin a) ->
            (\ (x : [a]b) ->
             f a x
             where
               /* Not recursive */
               f : {c} [c]b -> [a + c]b
               f = \{c} (y : [c]b) -> (#) a c b <> x y
               
             ) : [a]b -> [2 * a]b

Loading module Cryptol
Loading module test01
module test01
import Cryptol
/* Not recursive */
test01::a : {a, b} (fin a) => [a]b -> [2 * a]b
test01::a = \{a, b} (fin a) ->
            (\ (x : [a]b) ->
             f x
             where
               /* Not recursive */
               f : [a]b -> [a + a]b
               f = \ (y : [a]b) -> (#) a a b <> x y
               
             ) : [a]b -> [2 * a]b

